File: src/main/java/com/blogly/blogly/infrastructure/persistence/PostMapper.java
package com.blogly.blogly.infrastructure.persistence;

import com.blogly.blogly.domain.post.Content;
import com.blogly.blogly.domain.post.Post;
import com.blogly.blogly.domain.post.PostId;
import com.blogly.blogly.domain.post.Title;
import org.springframework.stereotype.Component;

@Component
public class PostMapper {

    public PostEntity toEntity(Post post) {
        if (post == null) {
            return null;
        }

        PostEntity entity = new PostEntity();

        entity.setId(post.getId().value());
        entity.setTitle(post.getTitle().value());
        entity.setContent(post.getContent().value());
        entity.setStatus(post.getStatus());

        return entity;
    }

    public Post toDomain(PostEntity entity) {
        if (entity == null) {
            return null;
        }

        return new Post(
                new PostId(entity.getId()),
                new Title(entity.getTitle()),
                new Content(entity.getContent()),
                entity.getStatus()
        );
    }
}

File: src/main/java/com/blogly/blogly/infrastructure/persistence/post/PostMapper.kt
package com.blogly.blogly.infrastructure.persistence.post

import com.blogly.blogly.domain.post.Content
import com.blogly.blogly.domain.post.Post
import com.blogly.blogly.domain.post.PostId
import com.blogly.blogly.domain.post.Title
import com.blogly.blogly.domain.user.UserId
import org.springframework.stereotype.Component

@Component
class PostMapper {

    fun toEntity(post: Post): PostEntity {
        val entity = PostEntity()

        entity.id = post.id.value
        entity.title = post.title.value
        entity.content = post.content.value
        entity.status = post.status
        entity.userId = post.userId.value

        return entity
    }

    fun toDomain(entity: PostEntity) =
        Post(
            Title(entity.title),
            Content(entity.content),
            UserId(entity.userId),
            PostId(entity.id),
            entity.status
        )
}
